### ISSUE TITLE
bad packet, are mods in use!? Couldn't read all bytes from a packet

### ISSUE THREAD

**Issue by SrMonsterYT** [LoneDev6/BungeePackFix#55]
this happens when I log into version 1.8, and then log into the 1.20.1 server

error 1: https://prnt.sc/SaWnZY-p6c7M

então ativei a opção Dwaterfall.packet-decode-logging=true

error 2: https://prnt.sc/_y7zw9d15RIU

BungeePackFix_1.1.5-beta-1
waterfall-1.20-562

removing bungeepackfix solves the problem and they are not kicked

**Comment by LoneDev6** [comment]
Just curious, but why are you logging in via 1.8 client on a 1.20.1 server?
I find it quite strange to do such thing as 1.8 can't render most of the 1.20 features.

**Comment by LoneDev6** [comment]
Also post your bungeecord log, do not post screenshots only.

**Comment by SrMonsterYT** [comment]
i'm using viaversion to support 1.8 - 1.20.4 i believe 80% of servers use it to do this.

the only error is this: 

[17:01:51] [Netty Worker IO Thread #4/WARN]: [/*:23340|SrMonsterYT] -> UpstreamBridge - bad packet, are mods in use!? Couldn't read all bytes from a packet. For more information, launch Waterfall with -Dwaterfall.packet-decode-logging=true
[17:01:51] [Netty Worker IO Thread #4/INFO]: [SrMonsterYT] disconnected with: DecoderException : net.md_5.bungee.protocol.BadPacketException: Couldn't read all bytes from a packet. For more information, launch Waterfall with -Dwaterfall.packet-decode-logging=true @ io.netty.handler.codec.MessageToMessageDecoder:98


no other errors except for the players who have joined who are in the image, now if you want the full log you will only see a normal server starting up

-Dwaterfall.packet-decode-logging=true

**Comment by LoneDev6** [comment]
> For more information, launch Waterfall with -Dwaterfall.packet-decode-logging=true

Did you run it with this Java argument enabled?
Check logs after you enable it and share them here.

**Comment by SrMonsterYT** [comment]
> > For more information, launch Waterfall with -Dwaterfall.packet-decode-logging=true
> 
> Did you run it with this Java argument enabled? Check logs after you enable it and share them here.

yes i did and the error changed to that: https://prnt.sc/_y7zw9d15RIU

it was the first thing I did to report the problem to you

**Comment by SrMonsterYT** [comment]
I left only these plugins and the error happens, when I remove the itemadder, the BungeePackFix error doesn't happen, it looks like a conflict

plugins: 

ViaVersion-4.9.3-SNAPSHOT.jar
ViaRewind-3.0.6-SNAPSHOT.jar
ViaBackwards-4.9.2-SNAPSHOT.jar
ProtocolLib.jar
item-nbt-api-plugin-2.12.1.jar

ItemsAdder 3.6.3-beta-9
LoneLibs 1.0.45

**Comment by SrMonsterYT** [comment]
@LoneDev6 sorry to tag you, but I've been waiting 1 week for a fix for this, it didn't happen in previous versions only in this version that it happens

**Comment by SrMonsterYT** [comment]
[21:58:18 WARN]: [/127.0.0.1:50217|SrMonsterYT] -> UpstreamBridge - A decoder exception has been thrown:
net.md_5.bungee.protocol.FastDecoderException: Error decoding packet class ServerboundResourcePackPacket with contents:
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 19 04 6e 75 6c 6c 03                            |..null.         |
+--------+-------------------------------------------------+----------------+
Caused by: net.md_5.bungee.protocol.BadPacketException: Packet GAME:TO_SERVER/25 (ServerboundResourcePackPacket) larger than expected, extra bytes: 5


I did a test on the local server and the problem is solved when I set the itemadder option:

apply-on-join: false

how to reproduce the problem just use BungeePackFix in version 1.1.5, and enter the server in 1.8, using viaversion, the server is in version 1.20.1

**Comment by SrMonsterYT** [comment]
@LoneDev6 it looks like the problem is in BungeePackFix and viaversion, I just saw more reports, and they point out that the problem is in bungeepackfix

image: https://prnt.sc/kKpmQT0v5rEY

please fix this problem
